sdk/nodejs/cmd/pulumi-language-nodejs/main.go

```diff
@@ -160,6 +160,31 @@ func newLanguageHost(nodePath, runPath, engineAddress,
 	}
 }
 
+func compatibleVersions(a, b semver.Version) (bool, string) {
+	switch {
+	case a.Major == 0 && b.Major == 0:
+		// If both major versions are pre-1.0, we require that the major and minor versions match.
+		if a.Minor != b.Minor {
+			return false, "Differing major or minor versions are not supported."
+		}
+
+	case a.Major >= 1 && b.Major >= 1:
+		// If both major versions are post-1.0, we require that the major versions match.
+		if a.Major != b.Minor {
+			return false, "Differing major versions are not supported."
+		}
+
+	case a.Major == 1 && b.Minor == 17 || b.Major == 1 && a.Minor == 17:
```

Comment:
Technically it feels like this needs an `&& b.Major == 0`, though that will only matter when we get to `2.17`?

Reply:
It doesn't: if we get to this case, we know that one version has a major version of zero and the other has a major version greater than zero.

Reply:
Added a clause for clarity.